addon/components/horizontal-bar-chart.js

```diff
@@ -435,28 +445,52 @@ const HorizontalBarChartComponent = ChartComponent.extend(FloatingTooltipMixin,
     // value and grouping labels on each side. Find the largest one on either
     // side.
     } else {
-      const positiveValues = this.get('positiveValues');
-      const negativeValues = this.get('negativeValues');
+      const minValue = this.get('minValue');
+      const maxValue = this.get('maxValue');
+      const maxLabelWidth = this.get('maxLabelWidth');
 
-      const positiveGroupingLabels = positiveValues.map((val) => {
+      const leftGroupingLabels = this.get('positiveValues').map((val) => {
         return this._getElementForValue(groupLabelElements, val);
       });
-      const positiveValueLabels = positiveValues.map((val) => {
-        return this._getElementForValue(valueLabelElements, val);
-      });
-      const negativeGroupingLabels = negativeValues.map((val) => {
+      const rightGroupingLabels = this.get('negativeValues').map((val) => {
         return this._getElementForValue(groupLabelElements, val);
       });
-      const negativeValueLabels = negativeValues.map((val) => {
-        return this._getElementForValue(valueLabelElements, val);
-      });
+      const maxLeftGroupingLabelWidth = d3.min([maxLabelWidth,
+        this._maxWidthOfElements(leftGroupingLabels)]);
+      const maxRightGroupingLabelWidth = d3.min([maxLabelWidth,
+        this._maxWidthOfElements(rightGroupingLabels)]);
+      const leftMostValueLabelWidth = this._getElementWidthForValue(valueLabelElements, minValue);
+      const rightMostValueLabelWidth = this._getElementWidthForValue(valueLabelElements, maxValue);
+
+      const padding = (2 * this.get('labelPadding')) + this.get('yAxisTitleHeightOffset') + 5;
+      const outerWidth = this.get('outerWidth');
+
+      const width = outerWidth - leftMostValueLabelWidth - rightMostValueLabelWidth - padding;
+      const xScale = this._xScaleForWidth(width);
+
+      const maxNegativeBarWidth = Math.abs(xScale(minValue) - xScale(0));
+      const maxPositiveBarWidth = Math.abs(xScale(maxValue) - xScale(0));
+
+      var leftWidth = leftMostValueLabelWidth;
+      if (maxLeftGroupingLabelWidth > maxNegativeBarWidth + leftMostValueLabelWidth) {
+        // In the case where the left grouping label is wider than the sum of the largest left bar
+        // and its value label, the amount of space to the left of the axis is fixed at the width
+        // of the grouping label. The amount of space to the right of most positive bar is also
+        // a fixed value label width. Knowing this, we can compute the width of the positive bar.
+        const realPositiveBarWidth = outerWidth - maxLeftGroupingLabelWidth - rightMostValueLabelWidth - padding;
+        // From the positive bar width, we can compute the negative bar width
+        const realNegativeBarWidth = this._mostNegativeBarWidth(realPositiveBarWidth);
+        leftWidth = maxLeftGroupingLabelWidth - realNegativeBarWidth;
```

Comment:
@philpee2 in scenarios when you shrink the container too far, the `realPositiveBarWidth` could be negative. That means, there's no way to fit all the content of the chart in the space. In these cases `realNegativeBarWidth` will come out to be negative and hence `leftWidth + rightWidth + padding > outerwidth`. 
Although there's no way to render the chart correctly, we can possibly set the upper bound of `leftWidth` as `outerwidth - rightWidth - padding`.